### ISSUE TITLE
Terminate End Event

### ISSUE THREAD

**Issue by ateodorescu** [phpmentors-jp/workflower#79]
Quotes from BPMN specification (Business Process Model and Notation (BPMN), v2.0.2):
- "If a token reaches a Terminate End Event, the entire Process is abnormally terminated." (chapter 13.5.6)
- "For a “terminate” End Event, the Process is abnormally terminated—no other ongoing Process instances are
affected." (page 443)
- "For a “terminate” End Event, the Sub-Process is abnormally terminated. In case of a multi-instance Sub-Process,
only the affected instance is terminated—no other ongoing Sub-Process instances or higher-level Sub-Process or
Process instances are affected." (chapter 13.5.6)

Basically when a token reaches the Terminate End Event all other existing tokens should be consumed and the active activities should be cancelled. This means that existing work items should be cancelled. If an activity has sub-processes then those process instances should be cancelled. The ProcessInstance could be marked an abnormally terminated.

**Comment by ateodorescu** [comment]
Fixed in https://github.com/phpmentors-jp/workflower/pull/81
